server/src/main/java/io/crate/metadata/doc/DocTableInfoFactory.java

```diff
@@ -21,188 +21,660 @@
+        Settings tableParameters;
+        IndexMetadata.State state;
+        int numberOfShards;
+        String[] concreteIndices;
+        try {
             concreteIndices = IndexNameExpressionResolver.concreteIndexNames(
                 metadata,
-                IndicesOptions.lenientExpandOpen(),
+                indexTemplateMetadata == null
+                    ? IndicesOptions.strictExpandOpen()
+                    : IndicesOptions.lenientExpandOpen(),
                 relation.indexNameOrAlias()
             );
+        } catch (IndexNotFoundException e) {
+            throw new RelationUnknown(relation.fqn(), e);
+        }
+        String[] concreteOpenIndices;
+        List<PartitionName> partitions;
+        if (indexTemplateMetadata == null) {
+            IndexMetadata index = metadata.index(relation.indexNameOrAlias());
+            if (index == null) {
+                if (concreteIndices.length > 0) {
+                    index = metadata.index(concreteIndices[0]);
+                    LOGGER.info(
+                        "Found indices={} for relation={} without index template. Orphaned partition?",
+                        concreteIndices,
+                        relation
+                    );
+                }
+                if (index == null) {
+                    throw new RelationUnknown(relation);
+                }
+            }
+            Settings indexSettings = index.getSettings();
+            tableParameters = indexSettings;
+            versionCreated = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(indexSettings);
+            versionUpgraded = indexSettings.getAsVersion(IndexMetadata.SETTING_VERSION_UPGRADED, null);
+            state = index.getState();
+            MappingMetadata mapping = index.mapping();
+            mappingSource = mapping == null ? Map.of() : mapping.sourceAsMap();
+            numberOfShards = index.getNumberOfShards();
+            concreteOpenIndices = concreteIndices;
+            if (concreteIndices.length == 0) {
+                throw new RelationUnknown(relation);
+            }
+            partitions = List.of();
+        } else {
+            mappingSource = XContentHelper.toMap(
+                indexTemplateMetadata.mapping().compressedReference(),
+                XContentType.JSON
+            );
+            mappingSource = Maps.getOrDefault(mappingSource, "default", mappingSource);
+            Settings indexSettings = indexTemplateMetadata.settings();
+            tableParameters = indexSettings;
+            versionCreated = indexSettings.getAsVersion(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT);
+            versionUpgraded = null;
+            boolean isClosed = Maps.getOrDefault(
+                Maps.getOrDefault(mappingSource, "_meta", Map.of()), "closed", false);
+            state = isClosed ? State.CLOSE : State.OPEN;
+            numberOfShards = indexSettings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 5);
```

Comment:
Can't we share this code for both cases as soon as we resolved the `indexSettings` (beside of the `state`)?

Reply:
Templates default the version_created to `Version.CURRENT`, tables don't because there is a concrete index.
For the shards it might be possible. I kept the logic closer to the existing behavior. I can follow up on this in a separate PR